rflx/pyrflx/typevalue.py

```diff
@@ -260,3 +460,350 @@ def accepted_type(self) -> type:
     @property
     def literals(self) -> Mapping[Name, Expr]:
         return {}
+
+    def assign(self, value: bytes, length: int = 0, check: bool = True) -> None:
+        """
+        :param value: Bitstring representation of the message
+        :param length: use length parameter if complete
+        length of the message is less than 1 Byte
+        :param check:
+        :return:
+        """
+
+        msg_as_bitstr: Bitstring = Bitstring().from_bytes(value)
+        self.assign_bitvalue(msg_as_bitstr, length)
+
+    def assign_bitvalue(self, value: Bitstring, length: int = 0) -> None:
+        """
+        :param value: Bitstring representation of the message
+        :param length: use length parameter if complete
+        length of the message is less than 1 Byte
+        """
+
+        current_field_name = self._next_field(model.INITIAL.name)
+        field_first_in_bitstr = 0
+        field_length = 0
+
+        while current_field_name != model.FINAL.name and (
+            field_first_in_bitstr + field_length
+        ) <= len(str(value)):
```

Comment:
(1) an Index Error would be raised
(2) nothing, the message would be parsed normally (I would like to keep it that way as this behaviour is very handy for parsing arrays of messages)

Reply:
(1) Would it make sense to raise an explicit exception for that case (a la "message is incomplete")? Is there a unit test for this case?
(2) Actually, this behavior is consistent to the SPARK parser - trailing bits of the input buffer are ignored while parsing. @senier: Do you see a problem there? Should it be checked that the parsed message ends exactly with the last bit of the buffer?

Reply:
I've had both cases in the past. IMHO we should not by default enforce that all bits are consumed by the parser. There could, however, be situations where you consider a message only valid if all bytes have been consumed (comparable to pyparsings `StringEnd`). I could not quickly see how/if *Kaitai Struct* and *Construct* handle this.

@treiher: I guess you could enforce that in your specification if the last field has an opaque type and a condition stating `Length = 0` and `Last => Message'Last`?

Reply:
@senier: I suppose you mean `First => Message'Last`, as `Last` is not a valid aspect in our specification. In theory this should work (although I think this workaround is quite ugly). In practice such a specification is rejected because `Message'Last` is not accepted as valid first expression during model verification. But there is an easier way to represent this constraint: adding a condition to the final node.

```
   ...
   Last_Field : Field_Type
      then null
         if Last_Field'Last = Message'Last;
```

Reply:
@treiher: Yes, that was the solution I came up with when trying the idea eventually. There still seems to be something wrong here, as I cannot prove it and my test case fails with Contraint_Error (discrimiant check failed). I'll open a separate issue on that.

Reply:
> (1) Would it make sense to raise an explicit exception for that case (a la "message is incomplete")? Is there a unit test for this case?

@rssen

Reply:
Added explicit error and unittest in 1a31ada